spark/src/main/scala/org/apache/spark/sql/delta/catalog/DeltaTableV2.scala

```diff
@@ -87,7 +87,27 @@ case class DeltaTableV2(
   // The loading of the DeltaLog is lazy in order to reduce the amount of FileSystem calls,
   // in cases where we will fallback to the V1 behavior.
   lazy val deltaLog: DeltaLog = {
-      DeltaLog.forTable(spark, rootPath, options)
+      val tableNameInException = catalogTable match {
+        case Some(ct) => Some(ct.identifier.copy(catalog = None).unquotedString)
+        case None => tableIdentifier
+      }
+
+      try {
+        DeltaLog.forTable(spark, rootPath, options)
```

Comment:
I added it to initialSnapshot, but I think it may not be needed, as it already calls the deltaLog.
Also, running "incompatible protocol change during the transaction" I'm not seeing it in stack trace. I'll invesitgate it more and come back.

The stack trace for that test:

<details>
[info] - incompatible protocol change during the transaction *** FAILED ***
[info]   org.apache.spark.sql.delta.InvalidProtocolVersionException: [DELTA_INVALID_PROTOCOL_VERSION] Delta protocol version is not supported by this version of Delta Lake: table "file:/sources/spark/spark-warehouse/org.apache.spark.sql.delta.DeltaProtocolVersionSuite/abc2" requires reader version 2147483647 and writer version 7, client supports reader versions 0, 1, 2, 3 and writer versions 0, 1, 2, 3, 4, 5, 7. Please upgrade to a newer release.
[info]   at org.apache.spark.sql.delta.DeltaLog.protocolCheck(DeltaLog.scala:409)
[info]   at org.apache.spark.sql.delta.DeltaLog.protocolRead(DeltaLog.scala:441)
[info]   at org.apache.spark.sql.delta.Snapshot.init(Snapshot.scala:170)
[info]   at org.apache.spark.sql.delta.Snapshot.<init>(Snapshot.scala:419)
[info]   at org.apache.spark.sql.delta.SnapshotManagement.$anonfun$createSnapshot$2(SnapshotManagement.scala:475)
[info]   at org.apache.spark.sql.delta.SnapshotManagement.createSnapshotFromGivenOrEquivalentLogSegment(SnapshotManagement.scala:600)
[info]   at org.apache.spark.sql.delta.SnapshotManagement.createSnapshotFromGivenOrEquivalentLogSegment$(SnapshotManagement.scala:588)
[info]   at org.apache.spark.sql.delta.DeltaLog.createSnapshotFromGivenOrEquivalentLogSegment(DeltaLog.scala:73)
[info]   at org.apache.spark.sql.delta.SnapshotManagement.createSnapshot(SnapshotManagement.scala:468)
[info]   at org.apache.spark.sql.delta.SnapshotManagement.createSnapshot$(SnapshotManagement.scala:461)
[info]   at org.apache.spark.sql.delta.DeltaLog.createSnapshot(DeltaLog.scala:73)
[info]   at org.apache.spark.sql.delta.SnapshotManagement.$anonfun$installLogSegmentInternal$1(SnapshotManagement.scala:768)
[info]   at org.apache.spark.sql.delta.SnapshotManagement.$anonfun$installLogSegmentInternal$1$adapted(SnapshotManagement.scala:760)
[info]   at scala.Option.map(Option.scala:230)
[info]   at org.apache.spark.sql.delta.SnapshotManagement.installLogSegmentInternal(SnapshotManagement.scala:760)
[info]   at org.apache.spark.sql.delta.SnapshotManagement.installLogSegmentInternal$(SnapshotManagement.scala:755)
[info]   at org.apache.spark.sql.delta.DeltaLog.installLogSegmentInternal(DeltaLog.scala:73)
[info]   at org.apache.spark.sql.delta.SnapshotManagement.$anonfun$updateInternal$1(SnapshotManagement.scala:751)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.recordFrameProfile(DeltaLogging.scala:140)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.recordFrameProfile$(DeltaLogging.scala:138)
[info]   at org.apache.spark.sql.delta.DeltaLog.recordFrameProfile(DeltaLog.scala:73)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.$anonfun$recordDeltaOperationInternal$1(DeltaLogging.scala:133)
[info]   at com.databricks.spark.util.DatabricksLogging.recordOperation(DatabricksLogging.scala:128)
[info]   at com.databricks.spark.util.DatabricksLogging.recordOperation$(DatabricksLogging.scala:117)
[info]   at org.apache.spark.sql.delta.DeltaLog.recordOperation(DeltaLog.scala:73)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.recordDeltaOperationInternal(DeltaLogging.scala:132)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.recordDeltaOperation(DeltaLogging.scala:122)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.recordDeltaOperation$(DeltaLogging.scala:112)
[info]   at org.apache.spark.sql.delta.DeltaLog.recordDeltaOperation(DeltaLog.scala:73)
[info]   at org.apache.spark.sql.delta.SnapshotManagement.updateInternal(SnapshotManagement.scala:746)
[info]   at org.apache.spark.sql.delta.SnapshotManagement.updateInternal$(SnapshotManagement.scala:745)
[info]   at org.apache.spark.sql.delta.DeltaLog.updateInternal(DeltaLog.scala:73)
[info]   at org.apache.spark.sql.delta.SnapshotManagement.$anonfun$update$3(SnapshotManagement.scala:697)
[info]   at org.apache.spark.sql.delta.DeltaLog.lockInterruptibly(DeltaLog.scala:165)
[info]   at org.apache.spark.sql.delta.SnapshotManagement.$anonfun$update$2(SnapshotManagement.scala:696)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.recordFrameProfile(DeltaLogging.scala:140)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.recordFrameProfile$(DeltaLogging.scala:138)
[info]   at org.apache.spark.sql.delta.DeltaLog.recordFrameProfile(DeltaLog.scala:73)
[info]   at org.apache.spark.sql.delta.SnapshotManagement.update(SnapshotManagement.scala:696)
[info]   at org.apache.spark.sql.delta.SnapshotManagement.update$(SnapshotManagement.scala:664)
[info]   at org.apache.spark.sql.delta.DeltaLog.update(DeltaLog.scala:73)
[info]   at org.apache.spark.sql.delta.OptimisticTransactionImpl.getNextAttemptVersion(OptimisticTransaction.scala:1792)
[info]   at org.apache.spark.sql.delta.OptimisticTransactionImpl.getNextAttemptVersion$(OptimisticTransaction.scala:1791)
[info]   at org.apache.spark.sql.delta.OptimisticTransaction.getNextAttemptVersion(OptimisticTransaction.scala:141)
[info]   at org.apache.spark.sql.delta.OptimisticTransactionImpl.$anonfun$checkForConflicts$1(OptimisticTransaction.scala:1740)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.recordFrameProfile(DeltaLogging.scala:140)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.recordFrameProfile$(DeltaLogging.scala:138)
[info]   at org.apache.spark.sql.delta.OptimisticTransaction.recordFrameProfile(OptimisticTransaction.scala:141)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.$anonfun$recordDeltaOperationInternal$1(DeltaLogging.scala:133)
[info]   at com.databricks.spark.util.DatabricksLogging.recordOperation(DatabricksLogging.scala:128)
[info]   at com.databricks.spark.util.DatabricksLogging.recordOperation$(DatabricksLogging.scala:117)
[info]   at org.apache.spark.sql.delta.OptimisticTransaction.recordOperation(OptimisticTransaction.scala:141)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.recordDeltaOperationInternal(DeltaLogging.scala:132)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.recordDeltaOperation(DeltaLogging.scala:122)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.recordDeltaOperation$(DeltaLogging.scala:112)
[info]   at org.apache.spark.sql.delta.OptimisticTransaction.recordDeltaOperation(OptimisticTransaction.scala:141)
[info]   at org.apache.spark.sql.delta.OptimisticTransactionImpl.checkForConflicts(OptimisticTransaction.scala:1738)
[info]   at org.apache.spark.sql.delta.OptimisticTransactionImpl.checkForConflicts$(OptimisticTransaction.scala:1730)
[info]   at org.apache.spark.sql.delta.OptimisticTransaction.checkForConflicts(OptimisticTransaction.scala:141)
[info]   at org.apache.spark.sql.delta.OptimisticTransactionImpl.$anonfun$doCommitRetryIteratively$4(OptimisticTransaction.scala:1571)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.recordFrameProfile(DeltaLogging.scala:140)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.recordFrameProfile$(DeltaLogging.scala:138)
[info]   at org.apache.spark.sql.delta.OptimisticTransaction.recordFrameProfile(OptimisticTransaction.scala:141)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.$anonfun$recordDeltaOperationInternal$1(DeltaLogging.scala:133)
[info]   at com.databricks.spark.util.DatabricksLogging.recordOperation(DatabricksLogging.scala:128)
[info]   at com.databricks.spark.util.DatabricksLogging.recordOperation$(DatabricksLogging.scala:117)
[info]   at org.apache.spark.sql.delta.OptimisticTransaction.recordOperation(OptimisticTransaction.scala:141)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.recordDeltaOperationInternal(DeltaLogging.scala:132)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.recordDeltaOperation(DeltaLogging.scala:122)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.recordDeltaOperation$(DeltaLogging.scala:112)
[info]   at org.apache.spark.sql.delta.OptimisticTransaction.recordDeltaOperation(OptimisticTransaction.scala:141)
[info]   at org.apache.spark.sql.delta.OptimisticTransactionImpl.$anonfun$doCommitRetryIteratively$3(OptimisticTransaction.scala:1569)
[info]   at scala.collection.immutable.Range.foreach$mVc$sp(Range.scala:158)
[info]   at org.apache.spark.sql.delta.OptimisticTransactionImpl.$anonfun$doCommitRetryIteratively$2(OptimisticTransaction.scala:1565)
[info]   at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.recordFrameProfile(DeltaLogging.scala:140)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.recordFrameProfile$(DeltaLogging.scala:138)
[info]   at org.apache.spark.sql.delta.OptimisticTransaction.recordFrameProfile(OptimisticTransaction.scala:141)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.$anonfun$recordDeltaOperationInternal$1(DeltaLogging.scala:133)
[info]   at com.databricks.spark.util.DatabricksLogging.recordOperation(DatabricksLogging.scala:128)
[info]   at com.databricks.spark.util.DatabricksLogging.recordOperation$(DatabricksLogging.scala:117)
[info]   at org.apache.spark.sql.delta.OptimisticTransaction.recordOperation(OptimisticTransaction.scala:141)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.recordDeltaOperationInternal(DeltaLogging.scala:132)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.recordDeltaOperation(DeltaLogging.scala:122)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.recordDeltaOperation$(DeltaLogging.scala:112)
[info]   at org.apache.spark.sql.delta.OptimisticTransaction.recordDeltaOperation(OptimisticTransaction.scala:141)
[info]   at org.apache.spark.sql.delta.OptimisticTransactionImpl.$anonfun$doCommitRetryIteratively$1(OptimisticTransaction.scala:1565)
[info]   at org.apache.spark.sql.delta.DeltaLog.lockInterruptibly(DeltaLog.scala:165)
[info]   at org.apache.spark.sql.delta.OptimisticTransactionImpl.lockCommitIfEnabled(OptimisticTransaction.scala:1541)
[info]   at org.apache.spark.sql.delta.OptimisticTransactionImpl.doCommitRetryIteratively(OptimisticTransaction.scala:1559)
[info]   at org.apache.spark.sql.delta.OptimisticTransactionImpl.doCommitRetryIteratively$(OptimisticTransaction.scala:1555)
[info]   at org.apache.spark.sql.delta.OptimisticTransaction.doCommitRetryIteratively(OptimisticTransaction.scala:141)
[info]   at org.apache.spark.sql.delta.OptimisticTransactionImpl.liftedTree1$1(OptimisticTransaction.scala:1064)
[info]   at org.apache.spark.sql.delta.OptimisticTransactionImpl.$anonfun$commitImpl$1(OptimisticTransaction.scala:992)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.recordFrameProfile(DeltaLogging.scala:140)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.recordFrameProfile$(DeltaLogging.scala:138)
[info]   at org.apache.spark.sql.delta.OptimisticTransaction.recordFrameProfile(OptimisticTransaction.scala:141)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.$anonfun$recordDeltaOperationInternal$1(DeltaLogging.scala:133)
[info]   at com.databricks.spark.util.DatabricksLogging.recordOperation(DatabricksLogging.scala:128)
[info]   at com.databricks.spark.util.DatabricksLogging.recordOperation$(DatabricksLogging.scala:117)
[info]   at org.apache.spark.sql.delta.OptimisticTransaction.recordOperation(OptimisticTransaction.scala:141)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.recordDeltaOperationInternal(DeltaLogging.scala:132)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.recordDeltaOperation(DeltaLogging.scala:122)
[info]   at org.apache.spark.sql.delta.metering.DeltaLogging.recordDeltaOperation$(DeltaLogging.scala:112)
[info]   at org.apache.spark.sql.delta.OptimisticTransaction.recordDeltaOperation(OptimisticTransaction.scala:141)
[info]   at org.apache.spark.sql.delta.OptimisticTransactionImpl.commitImpl(OptimisticTransaction.scala:989)
[info]   at org.apache.spark.sql.delta.OptimisticTransactionImpl.commitImpl$(OptimisticTransaction.scala:984)
[info]   at org.apache.spark.sql.delta.OptimisticTransaction.commitImpl(OptimisticTransaction.scala:141)
[info]   at org.apache.spark.sql.delta.OptimisticTransactionImpl.commit(OptimisticTransaction.scala:960)
[info]   at org.apache.spark.sql.delta.OptimisticTransactionImpl.commit$(OptimisticTransaction.scala:957)
[info]   at org.apache.spark.sql.delta.OptimisticTransaction.commit(OptimisticTransaction.scala:141)
[info]   at org.apache.spark.sql.delta.DeltaProtocolVersionSuiteBase.$anonfun$$init$$3(DeltaProtocolVersionSuite.scala:532)
[info]   at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[info]   at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally(SparkErrorUtils.scala:64)
[info]   at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally$(SparkErrorUtils.scala:61)
[info]   at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:94)
[info]   at org.apache.spark.sql.test.SQLTestUtilsBase.withTable(SQLTestUtils.scala:306)
[info]   at org.apache.spark.sql.test.SQLTestUtilsBase.withTable$(SQLTestUtils.scala:304)
[info]   at org.apache.spark.sql.delta.DeltaProtocolVersionSuite.withTable(DeltaProtocolVersionSuite.scala:3207)
[info]   at org.apache.spark.sql.delta.DeltaProtocolVersionSuiteBase.$anonfun$$init$$2(DeltaProtocolVersionSuite.scala:518)
[info]   at org.apache.spark.sql.delta.DeltaProtocolVersionSuiteBase.$anonfun$$init$$2$adapted(DeltaProtocolVersionSuite.scala:512)
[info]   at scala.collection.immutable.List.foreach(List.scala:431)
[info]   at org.apache.spark.sql.delta.DeltaProtocolVersionSuiteBase.$anonfun$$init$$1(DeltaProtocolVersionSuite.scala:512)
[info]   at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[info]   at org.scalatest.enablers.Timed$$anon$1.timeoutAfter(Timed.scala:127)
[info]   at org.scalatest.concurrent.TimeLimits$.failAfterImpl(TimeLimits.scala:282)
[info]   at org.scalatest.concurrent.TimeLimits.failAfter(TimeLimits.scala:231)
[info]   at org.scalatest.concurrent.TimeLimits.failAfter$(TimeLimits.scala:230)
[info]   at org.apache.spark.SparkFunSuite.failAfter(SparkFunSuite.scala:69)
[info]   at org.apache.spark.SparkFunSuite.$anonfun$test$2(SparkFunSuite.scala:155)
[info]   at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
[info]   at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
[info]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:22)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:20)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike$$anon$1.apply(AnyFunSuiteLike.scala:226)
[info]   at org.apache.spark.SparkFunSuite.withFixture(SparkFunSuite.scala:227)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike.invokeWithFixture$1(AnyFunSuiteLike.scala:224)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike.$anonfun$runTest$1(AnyFunSuiteLike.scala:236)
[info]   at org.scalatest.SuperEngine.runTestImpl(Engine.scala:306)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike.runTest(AnyFunSuiteLike.scala:236)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike.runTest$(AnyFunSuiteLike.scala:218)
[info]   at org.apache.spark.SparkFunSuite.org$scalatest$BeforeAndAfterEach$$super$runTest(SparkFunSuite.scala:69)
[info]   at org.scalatest.BeforeAndAfterEach.runTest(BeforeAndAfterEach.scala:234)
[info]   at org.scalatest.BeforeAndAfterEach.runTest$(BeforeAndAfterEach.scala:227)
[info]   at org.apache.spark.SparkFunSuite.runTest(SparkFunSuite.scala:69)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike.$anonfun$runTests$1(AnyFunSuiteLike.scala:269)
[info]   at org.scalatest.SuperEngine.$anonfun$runTestsInBranch$1(Engine.scala:413)
[info]   at scala.collection.immutable.List.foreach(List.scala:431)
[info]   at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:401)
[info]   at org.scalatest.SuperEngine.runTestsInBranch(Engine.scala:396)
[info]   at org.scalatest.SuperEngine.runTestsImpl(Engine.scala:475)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike.runTests(AnyFunSuiteLike.scala:269)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike.runTests$(AnyFunSuiteLike.scala:268)
[info]   at org.scalatest.funsuite.AnyFunSuite.runTests(AnyFunSuite.scala:1564)
[info]   at org.scalatest.Suite.run(Suite.scala:1114)
[info]   at org.scalatest.Suite.run$(Suite.scala:1096)
[info]   at org.scalatest.funsuite.AnyFunSuite.org$scalatest$funsuite$AnyFunSuiteLike$$super$run(AnyFunSuite.scala:1564)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike.$anonfun$run$1(AnyFunSuiteLike.scala:273)
[info]   at org.scalatest.SuperEngine.runImpl(Engine.scala:535)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike.run(AnyFunSuiteLike.scala:273)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike.run$(AnyFunSuiteLike.scala:272)
[info]   at org.apache.spark.SparkFunSuite.org$scalatest$BeforeAndAfterAll$$super$run(SparkFunSuite.scala:69)
[info]   at org.scalatest.BeforeAndAfterAll.liftedTree1$1(BeforeAndAfterAll.scala:213)
[info]   at org.scalatest.BeforeAndAfterAll.run(BeforeAndAfterAll.scala:210)
[info]   at org.scalatest.BeforeAndAfterAll.run$(BeforeAndAfterAll.scala:208)
[info]   at org.apache.spark.SparkFunSuite.run(SparkFunSuite.scala:69)
[info]   at org.scalatest.tools.Framework.org$scalatest$tools$Framework$$runSuite(Framework.scala:321)
[info]   at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:517)
[info]   at sbt.ForkMain$Run.lambda$runTest$1(ForkMain.java:413)
[info]   at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[info]   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[info]   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[info]   at java.lang.Thread.run(Thread.java:750)
</details>

Reply:
> I added it to initialSnapshot, but I think it may not be needed, as it already calls the deltaLog

The case I worried about was when some _other_ client upgrades protocol after we already have a `DeltaLog` reference with a snapshot, e.g. 
```scala
val deltaLog = DeltaLog.forTable(...) // creates a log with a snapshot at v1
// some other client commits an incompatible protocol change
deltaLog.update() // encounters protocol change while moving to v2
```
I'm not sure how to unit test that tho -- by definition all DeltaLog instances understand the same protocol versions.

Reply:
Translating the above to `DeltaTableV2`:
* The first time we create a DTV2 on a given client, `DeltaLog.forTable` will create a new instance, which requires creating a snapshot, which will potentially trigger protocol error. 
* The second (and subsequent) DTV2 creation on that same client, `DeltaLog.forTable` will hit in cache and no snapshot will be created or updated at that time.

Either way, the subsequent trigger of `initialSnapshot` will call `deltaLog.update`, which could trigger a protocol error if somebody else upgraded the protocol since we last updated the snapshot.

Reply:
Update: The unit tests have already solved how to write an incompatible version. So we can test the `initialSnapshot` path by:
```scala
// Create the table
spark.range(1).write.format("delta").save(path.getCanonicalPath)

// capture a DeltaLog instance with a snapshot at v1
val (deltaLog, snapshot) = DeltaLog.forTableWithSnapshot(...)

// NOTE: Needs to create a new commit for existing table, rather than clobbering commit 0
overwriteDeltaLogWithVersion(..., snapshot.version + 1)

// Will not create a new DeltaLog (we already have one), but snapshot update will fail
val exceptionRead = intercept[InvalidProtocolVersionException] {
  spark.read.format("delta").table(protocolTableName)
}
```

Reply:
Thanks for detailed explanation. I created a unit test, and can confirm without the withEnrichedInvalidProtocolVersionException wrapper in initialSnapshot, the error message is not enhanced.

Adding the unit test in my next commit